src/store/participantsStore.js

```diff
@@ -640,6 +664,47 @@ const actions = {
 		}
 	},
 
+	/**
+	 * Update participant in store according to a new participant object
+	 *
+	 * @param {object} context store context
+	 * @param {object} data the wrapping object;
+	 * @param {string} data.token the conversation token;
+	 * @param {object} data.participant the new participant object;
+	 * @param {boolean} data.hasUserStatuses whether user status is enabled or not;
+	 * @return {boolean} whether the participant was changed
+	 */
+	updateParticipantIfHasChanged(context, { token, participant, hasUserStatuses }) {
+		const { attendeeId } = participant
+		const oldParticipant = context.state.attendees[token][attendeeId]
+
+		// Update, if status has changed
+		if (hasUserStatuses && (oldParticipant.status !== participant.status
+			|| oldParticipant.statusMessage !== participant.statusMessage
+			|| oldParticipant.statusIcon !== participant.statusIcon
+			|| oldParticipant.statusClearAt !== participant.statusClearAt
+		)) {
+			context.commit('updateParticipant', { token, attendeeId, updatedData: participant })
+			emitUserStatusUpdated(participant)
+			return true
+		}
+
+		// Check if any property were changed (no properties except status-related supposed to be added or deleted)
+		for (const key of Object.keys(participant)) {
+			// "sessionIds" is the only property with non-primitive (array) value and cannot be compared by ===
+			if (key !== 'sessionIds' && oldParticipant[key] !== participant[key]) {
+				context.commit('updateParticipant', { token, attendeeId, updatedData: participant })
+				return true
+			} else if (key === 'sessionIds' && (oldParticipant.sessionIds.length !== participant.sessionIds.length
+				|| !oldParticipant.sessionIds.every((element, index) => element === participant.sessionIds[index]))) {
+				context.commit('updateParticipant', { token, attendeeId, updatedData: participant })
+				return true
+			}
+		}
```

Comment:
Might be a bit better in performance, but i worried about order in the array. Although for joined participants we'll update them on previous step, because of `lastPing`, so should be fine

Reply:
Yeah, I also had that thought. We can transform it into a set and compare maybe ? 
Perhaps the previous suggestion is enough too but I haven't checked.

Reply:
> but i worried about order in the array

The current check also rely on the order because it checks by `[index]`